src/httpserver.cpp

```diff
@@ -438,6 +430,8 @@ bool InitHTTPServer()
     workQueue = new WorkQueue<HTTPClosure>(workQueueDepth);
     eventBase = base;
     eventHTTP = http;
+    base_ctr.release();
+    http_ctr.release();
```

Comment:
Regarding purpose: the one thing this brings is that no matter how things end, the base and http objects will be properly cleaned up. I have no idea what kind of side-effects an initialized and forgotten event base or evhttp might have. Perhaps none whatsoever.